hbase-replication/src/main/java/org/apache/hadoop/hbase/replication/ReplicationTrackerZKImpl.java

```diff
@@ -190,6 +193,7 @@ private boolean refreshOtherRegionServersList(boolean watch) {
     } catch (KeeperException e) {
       this.abortable.abort("Get list of registered region servers", e);
     }
-    return result;
+    return result == null ? null :
+      result.stream().map(ServerName::parseServerName).collect(Collectors.toList());
```

Comment:
Let's take this opportunity to never return null here. If result is null, let's return `emptyList()`

Reply:
Thank @virajjasani for reviwing.
For method `refreshOtherRegionServersList` that calls `getRegisteredRegionServers`, it's different that getting null and empty list. It returns true or false depending on whether the result is null. Getting `null` means that an exception has occurred or the node does not exist in ZK.
So it is not very recommended to make this change, at least in this issue.

Reply:
Ah, I wish we could fix that boolean condition too, but I agree that it doesn't need to be part of this PR. Maybe later sometime.
Thanks